connect-button/src/main/java/com/ifttt/connect/ui/ConnectButton.java

```diff
@@ -459,13 +460,24 @@ private UserTokenAsyncTask(CredentialsProvider callback, UserTokenCallback userT
 
         @Override
         protected String doInBackground(Void... voids) {
-            return callback.getUserToken();
+            try {
+                return callback.getUserToken();
+            } catch (Exception e) {
+                if (BuildConfig.DEBUG) {
+                    e.printStackTrace();
+                }
+
+                return null;
```

Comment:
Can the sdk work correctly with a null `userToken`? 

When the `ConnectButton` needs to fetch the connection to display correct UI according to status, it will need the user token?

Reply:
I think this should be the correct setup. Currently we don't have anything in place to account for crashes from the implementation. This would cause the host app to crash when we try to fetch the user token. 

The ConnectButton doesn't necessarily need the `userToken` to render UI, without the token it will just render the initial state, which we probably cannot do anything else if we are handling a crashed getUserToken method call.